yggdrasil/services/rook-ceph/ceph/ceph.yaml

```
@@ -4,3 +4,63 @@ rook-ceph-cluster:
       {{- if .Values.registry }}
       image: {{ .Values.registry }}/ceph/ceph:v16.2.5
       {{- end }}
+    {{- if .Values.enableCephAKS }}
+    mon: 
+      volumeClaimTemplate:
+        spec:
+          storageClassName: managed-premium
+          resources:
+            requests:
+              storage: 10Gi
+    storage:
+      storageClassDeviceSets:
+      - name: set1
+      # This is the number of OSDs that will be created.
+        count: 4 
+        portable: true
+        placement:
+          tolerations:
+          - key: storage-node
+            operator: Exists
+          nodeAffinity:
+            requiredDuringSchedulingIgnoredDuringExecution:
+              nodeSelectorterms:
+              - matchExpressions:
+                - key: agentpool
+                  operator: In
+                  values:
+                  - npstorage
+          podAntiAffinity:
+            preferredDuringSchedulingIgnoredDuringExecution:
+            - weight: 100
+              podAffinityTerm:
+                labelSelector:
+                  matchExpressions:
+                  - key: app
+                    operator: In
+                    values:
+                    - rook-ceph-osd
+                  - key: app
+                    operator: In
+                    values:
+                    - rook-ceph-osd-prepare
```

Maybe it would be nicer to replace these `matchExpressions` with [`matchLabels`](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.22/#labelselector-v1-meta), e.g.
```suggestion
                  matchLabels:
                    app: rook-ceph-osd
                    app: rook-ceph-osd-prepare
```
Now, looking at the suggestion I just made, it is clear that that will not work... `app` can not have two values. We need to rework this as it is also a problem in the current form: the `matchExpressions` are ANDed (see above link). Basically, right now it does nothing because it will not match any pod.

---

Well, I think the solution is
```suggestion
                  matchExpressions:
                  - key: app
                    operator: In
                    values:
                    - rook-ceph-osd
                    - rook-ceph-osd-prepare
```

---

Looking again I think the solution is simply:
```suggestion
                  matchExpressions:
                  - key: app
                    operator: In
                    values:
                    - rook-ceph-osd
                    - rook-ceph-osd-prepare
```

---

I have implemented this change in my latest commit.